src/lib.rs

```diff
@@ -91,28 +91,22 @@ impl Lockfile {
     /// Use this instead of the destructor when you want to see if any errors occured when
     /// removing the file.
     pub fn release(mut self) -> Result<(), io::Error> {
-        self.close_file();
+        // Closes the file.
+        self.handle.take().expect("handle already dropped");
```

Comment:
Should have an explicit drop here to ensure file is closed before removed. Otherwise I think the compiler will move the drop to the end of the scope.

Reply:
I don't think so, see line 870 of the assembly at https://rust.godbolt.org/z/dQN9co

Reply:
@TyPR124 drop timings are well defined in Rust. If you don't bind a value to a variable (or you do, but the place you bind it is `_`), then the value is immediately dropped. Bindings are dropped in reverse-declaration order at end of scope, but unbound things are eagerly dropped.

Reply:
@CAD97, that seems counterintuitive, why would binding something to `_` be the same as not binding it to anything?

Reply:
Okay, I guess I was wrong, though I can't find it explicitly documented anywhere (just that drop happens when it goes "out of scope" which I thought would be after the call to remove, but I guess a semicolon can end a scope too?)

Reply:
@dunmatt the reason is that `_` isn't a variable name. It's not actually a binding; it's just an explicit acknowledgement that the value isn't needed and can be thrown away. (If you want an unused binding, you have to give it a name, like `_guard`.)

It may seem useless or problematic when you only consider `let _ = ...`; why not just not bind it then? But for more complicated patterns, like `let Struct { left, right: _ } = ...` or `let (one, _, three) = ...`, it allows you to say "I don't need this, throw it away".

`_` is not a variable binding, it's more like a trash can.

Reply:
@CAD97 @dunmatt It does appear that `let _ = ...` is different than no binding at all, at least for drop ordering. See https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=84d4e14297dbc582bf1d4dc99a046c55

Reply:
For some reason the `let _ = d;` isn't moving the `d` here. My information is correct when the value on the right hand side is a temporary (i.e. doesn't already have a named place), but this seems to be an ugly edge case that I would _not_ have expected.

So now it's a Twitter poll and hopefully sometime who knows why it's like this will see it: https://twitter.com/CAD97_/status/1230002045903671298?s=19

Reply:
This was instructive for me - I did not understand how drop orders work in this level of detail.